src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -620,6 +648,21 @@ private Task ProcessWindowUpdateFrameAsync()
                 throw new Http2ConnectionErrorException(CoreStrings.Http2ErrorWindowUpdateIncrementZero, Http2ErrorCode.PROTOCOL_ERROR);
             }
 
+            if (_incomingFrame.StreamId == 0)
+            {
+                if (!_frameWriter.TryUpdateConnectionWindow(_incomingFrame.WindowUpdateSizeIncrement))
+                {
+                    throw new Http2ConnectionErrorException(CoreStrings.Http2ErrorWindowUpdateSizeInvalid, Http2ErrorCode.FLOW_CONTROL_ERROR);
+                }
+            }
+            else if (_streams.TryGetValue(_incomingFrame.StreamId, out var stream))
+            {
+                if (!stream.TryUpdateOutputWindow(_incomingFrame.WindowUpdateSizeIncrement))
+                {
+                    throw new Http2StreamErrorException(_incomingFrame.StreamId, CoreStrings.Http2ErrorWindowUpdateSizeInvalid, Http2ErrorCode.FLOW_CONTROL_ERROR);
+                }
+            }
```

Comment:
At least leave a comment that window updates for closed streams are ignored due to a race condition between the client receiving all the data an the steam closing.

Reply:
Done.